platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -423,6 +423,42 @@ static uint32_t esp_block_len_to_mask(uint32_t block_len)
 	return block_len - 1;
 }
 
+/* Return supported power-of-two AR window size */
+static uint32_t ipsec_antireplay_ws(uint32_t win_size)
+{
+	if (win_size & (win_size - 1)) {
+		win_size--;
+		win_size = (win_size | win_size >> 1);
+		win_size = (win_size | win_size >> 2);
+		win_size = (win_size | win_size >> 4);
+		win_size = (win_size | win_size >> 8);
+		win_size++;
+	}
+
+	return win_size;
+}
+
+/* AR window management initialization */
+static void ipsec_antireplay_init(ipsec_sa_t *ipsec_sa,
+				  const odp_ipsec_sa_param_t *param)
+{
+	uint16_t num_bkts = 0;
+
+	ipsec_sa->ar.win_size =
+		ipsec_antireplay_ws(param->inbound.antireplay_ws);
+
+	num_bkts = (ipsec_sa->ar.win_size / IPSEC_ARWIN_BUCKET_SIZE) + 1;
+	if (num_bkts < IPSEC_ARWIN_BUCKET_MIN)
+		num_bkts = IPSEC_ARWIN_BUCKET_MIN;
+
+	ipsec_sa->ar.nb_bucket = num_bkts;
+	ipsec_sa->hot.in.wintop_seq = 0;
+	memset(ipsec_sa->ar.bucket_arr, 0,
+	       sizeof(uint64_t) * IPSEC_ARWIN_BUCKET_MAX);
```

Comment:
This could clear only the part of the array actually used.

Reply:
ACK, will make a change to only clear the required number of buckets(ipsec_sa->ar.nb_bucket) for that particular window size.